code/modules/projectiles/guns/ballistic.dm

```diff
@@ -236,6 +249,30 @@
 	..()
 	load_ammo(A, user)
 
+	if(suppressible)
+		if(istype(A, /obj/item/silencer))
+			if(!user.is_holding(src))	//if we're not in his hands
+				to_chat(user, "<span class='notice'>You'll need [src] in your hands to do that.</span>")
+				return CLICKCHAIN_DO_NOT_PROPAGATE
+			if(!user.attempt_insert_item_for_installation(A, src))
+				return CLICKCHAIN_DO_NOT_PROPAGATE
+			to_chat(user, "<span class='notice'>You screw [A] onto [src].</span>")
+			silenced = TRUE
+			w_class = ITEMSIZE_NORMAL
+			update_icon()
+			return CLICKCHAIN_DO_NOT_PROPAGATE
+		else if(istype(A, /obj/item/tool/wrench))
+			var/obj/item/silencer/S = new (get_turf(user))
+			if(silenced)
```

Comment:
you're making a new silencer everytime someone uses a wrench on a gun that can be suppressed